### ISSUE TITLE
Output <iteration.XML> every N iterations

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU#218]
Outputting state to disk is expensive, but useful.

In some cases it may not be worth exporting the full state each iteration, but rather every `N` iterations. 

I.e. Periodic check-pointing

As of 1.5.0 we are introducing an additional command line parameter for outputting to disk. 

Currently a value of `0` indicates no output, whilst any other value indicates we want output.
It would make sense that we use this variable to output every N iterations. I.e. 

```
../../bin/linux-x64/Release_Console/GameOfLife iterations/0.xml 1000 0 50
```
would run the GameOfLife Release_Console executable against `iterations/0.xml` for `1000` iterations, using device `0` and output an XML file every `50` iterations, producing `50.xml`, `100.xml`, `150.xml` etc.

**Comment by ptheywood** [comment]
Implemented in abce35e

Updated `-h|--help` output
```
../../bin/linux-x64/Release_Console/GameOfLife -h
FLAMEGPU Console mode

usage: GameOfLife [-h] [--help] input_path num_iterations [cuda_device_id] [XML_output_override]

required arguments:
  input_path           Path to initial states XML file OR path to output XML directory
  num_iterations       Number of simulation iterations

options arguments:
  -h, --help           Output this help message.
  cuda_device_id       CUDA device ID to be used. Default is 0.
  XML_output_frequency Frequency of XML output
                         0 = No output
                         1 = Every 1 iteration
                         5 = Every 5 iterations
                         Default value: 1
```

The final argument now accepts integer values, but also outputs the final iteration if it is not a multiple. 

I.e. 

```
../../bin/linux-x64/Release_Console/GameOfLife iterations/0.xml 25 0 6
FLAMEGPU Console mode
Initial states: iterations/0.xml
Output dir: iterations/
GPU 0: TITAN X (Pascal), SM61, Linux, pciBusId 2
64Bit System Detected
Allocating Host and Device memory
Processing Simulation Step 1
Processing Simulation Step 2
Processing Simulation Step 3
Processing Simulation Step 4
Processing Simulation Step 5
Processing Simulation Step 6
Iteration 6 Saved to XML
Processing Simulation Step 7
Processing Simulation Step 8
Processing Simulation Step 9
Processing Simulation Step 10
Processing Simulation Step 11
Processing Simulation Step 12
Iteration 12 Saved to XML
Processing Simulation Step 13
Processing Simulation Step 14
Processing Simulation Step 15
Processing Simulation Step 16
Processing Simulation Step 17
Processing Simulation Step 18
Iteration 18 Saved to XML
Processing Simulation Step 19
Processing Simulation Step 20
Processing Simulation Step 21
Processing Simulation Step 22
Processing Simulation Step 23
Processing Simulation Step 24
Iteration 24 Saved to XML
Processing Simulation Step 25
Iteration 25 Saved to XML
Total Processing time: 153.156326 (ms)
```

```
ls iterations/ | sort -n
0.xml
6.xml
12.xml
18.xml
24.xml
25.xml
```

**Comment by ptheywood** [comment]
For the example GameOfLife Model, 100 iterations with no XML output takes `9.44ms` on a Titan X(Pascal) using linux.

With XML output every iteration it takes ~ `2950ms`. Increasing the number of iterations between outputs shows the following performance behaviour with this model:

![gameoflifexmloutputperiod](https://user-images.githubusercontent.com/628937/40776100-b4278070-64c1-11e8-828f-ed316b183912.png)
(Excuse the wiggly line, Libre Office Calc's Export to Image does not seem to agree with curved lines)
